python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+
+        fitParams = fitResult.best_values
+        if fitParams['flux'] <= 1.:   # usually around 0.1 -- the minimun flux allowed -- i.e. bad fit.
+            out = DipoleFitAlgorithm.resultsOutput(
+                np.nan, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan, np.nan,
+                np.nan, np.nan, np.nan)
+            if return_fitObj:  # for debugging
+                return out, fitResult
+            return out
+
+        centroid = ((fitParams['xcenPos'] + fitParams['xcenNeg']) / 2.,
+                    (fitParams['ycenPos'] + fitParams['ycenNeg']) / 2.)
+        dx, dy = fitParams['xcenPos'] - fitParams['xcenNeg'], fitParams['ycenPos'] - fitParams['ycenNeg']
+        angle = np.arctan2(dy, dx) / np.pi * 180.   # convert to degrees (should keep as rad?)
+
+        # Exctract flux value, compute signalToNoise from flux/variance_within_footprint
+        # Also extract the stderr of flux estimate.
+        def computeSumVariance(exposure, footprint):
+            box = footprint.getBBox()
+            subim = afwImage.MaskedImageF(exposure.getMaskedImage(), box, afwImage.PARENT)
+            return np.sqrt(np.nansum(subim.getArrays()[1][:, :]))
+
+        fluxVal = fluxVar = fitParams['flux']
+        fluxErr = fluxErrNeg = fitResult.params['flux'].stderr
+        if self.posImage is not None:
+            fluxVar = computeSumVariance(self.posImage, source.getFootprint())
+        else:
+            fluxVar = computeSumVariance(self.diffim, source.getFootprint())
+
+        fluxValNeg, fluxVarNeg = fluxVal, fluxVar
+        if separateNegParams:
+            fluxValNeg = fitParams['fluxNeg']
+            fluxErrNeg = fitResult.params['fluxNeg'].stderr
+        if self.negImage is not None:
+            fluxVarNeg = computeSumVariance(self.negImage, source.getFootprint())
+
+        try:
+            signalToNoise = np.sqrt((fluxVal/fluxVar)**2 + (fluxValNeg/fluxVarNeg)**2)
+        except:  # catch divide by zero - should never happen.
+            signalToNoise = np.nan
+
+        out = DipoleFitAlgorithm.resultsOutput(
+            fitParams['xcenPos'], fitParams['ycenPos'], fitParams['xcenNeg'], fitParams['ycenNeg'],
+            fluxVal, -fluxValNeg, fluxErr, fluxErrNeg, centroid[0], centroid[1], angle,
+            signalToNoise, fitResult.chisqr, fitResult.redchi)
+
+        if return_fitObj:  # for debugging
+            return out, fitResult
+        return out
+
+    def displayFitResults(self, footprint, result):
+        """Display data, model fits and residuals (currently uses matplotlib display functions).
+        @param footprint Footprint containing the dipole that was fit
+        @param result `lmfit.MinimizerResult` object returned by `lmfit` optimizer
+        """
+
+        from . import utils as ipUtils
+        plt = ipUtils.importMatplotlib()
+        if not plt:  # Exit silently if no matplotlib installed
+            return
```

Comment:
I suggest this fail if you can't import matplotlib -- again, no need for a function, I suspect